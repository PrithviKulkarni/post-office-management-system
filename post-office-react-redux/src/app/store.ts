import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import letterReducer from '../features/LetterSlice';
import parcelReducer from '../features/ParcelSlice';
import depotReducer from '../features/DepotSlice';
import driverReducer from '../features/DriverSlice' 
export const store = configureStore({
  reducer: {
    letter: letterReducer,
    parcel: parcelReducer,
    depot: depotReducer,
    driver: driverReducer
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
