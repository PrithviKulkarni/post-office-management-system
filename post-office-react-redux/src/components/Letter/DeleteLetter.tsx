import React from "react";
import Card from "react-bootstrap/Card";
import { useParams, useNavigate } from "react-router-dom";
import { useAppDispatch, useAppSelector } from "../../app/hooks";
import { RootState } from "../../app/store";
import { getLetterById, deleteALetter } from "../../features/LetterSlice";
import Button from "react-bootstrap/Button";
import { Letter } from '../interfaces/letterInterface';

/**
 * 
 * @returns a new page with the specified chat data to be displayed and confirmed for deletion.
 */
const DeleteLetter = () => {
  const params = useParams();
  const dispatch = useAppDispatch();
  const navigate = useNavigate();
  const letter : Letter | undefined = useAppSelector((state: RootState) =>
    getLetterById(state, params.letterId!)
  );

  const handleDelete = () => {
    dispatch(deleteALetter(letter!));
    navigate("/letters")
  };

  return (
    <div style={{
      alignItems: "center",
      display: "flex",
      flexDirection: "column",
    }}
    data-testid="deleteLetter"
    className="deleteLetter">
    <Card
          style={{
            width: "25rem",
            padding: "1rem",
            margin: "5px",
            backgroundColor: "rgb(40,40,40)",
            display: "flex",
            justifyContent: "center",
            alignItems: "center",
          }}
          key={letter?.id}
        >
          <Card.Body>
            <Card.Title style={{ color: "white", fontWeight: "bold" }}>
              {letter?.address}
            </Card.Title>
            <Card.Subtitle className="mb-2 text-muted">
              {letter?.dispatched.toString().slice(0, 24)}
            </Card.Subtitle>
            <Card.Text style={{ color: "white" }}>
              {letter?.delivery_class}
            </Card.Text>
            <Card.Subtitle className="mb-2 text-muted">
              {letter?.delivered ? "Delivered" : "Not Delivered" }
            </Card.Subtitle>
          </Card.Body>
          <Button onClick={handleDelete} data-testid="deleteButton">Confirm Delete</Button>
        </Card>
    </div>
  );
};

export default DeleteLetter;
