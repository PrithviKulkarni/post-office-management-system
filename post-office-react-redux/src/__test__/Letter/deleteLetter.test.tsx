import React from "react";
import { render, cleanup, screen } from "@testing-library/react";
import { store } from "../../app/store";
import { Provider } from "react-redux";
import DeleteLetter from "../../components/Letter/DeleteLetter";
import { BrowserRouter } from "react-router-dom";

afterEach(cleanup);
it("renders Delete Letter page correctly", () => {
  const { asFragment } = render(
    <Provider store={store}>
      <BrowserRouter>
        <DeleteLetter />
      </BrowserRouter>
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("Verifies the Delete Letter render", () => {
  render(
    <Provider store={store}>
      <BrowserRouter>
        <DeleteLetter />
      </BrowserRouter>
    </Provider>
  );

  expect(screen.getByTestId("deleteLetter")).toHaveClass("deleteLetter");
});
