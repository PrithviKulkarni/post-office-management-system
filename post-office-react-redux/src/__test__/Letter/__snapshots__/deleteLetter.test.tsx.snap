// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Delete Letter page correctly 1`] = `
<DocumentFragment>
  <div
    class="deleteLetter"
    data-testid="deleteLetter"
    style="align-items: center; display: flex; flex-direction: column;"
  >
    <div
      class="card"
      style="width: 25rem; padding: 1rem; margin: 5px; background-color: rgb(40, 40, 40); display: flex; justify-content: center; align-items: center;"
    >
      <div
        class="card-body"
      >
        <div
          class="card-title h5"
          style="color: white; font-weight: bold;"
        />
        <div
          class="mb-2 text-muted card-subtitle h6"
        />
        <p
          class="card-text"
          style="color: white;"
        />
        <div
          class="mb-2 text-muted card-subtitle h6"
        >
          Not Delivered
        </div>
      </div>
      <button
        class="btn btn-primary"
        data-testid="deleteButton"
        type="button"
      >
        Confirm Delete
      </button>
    </div>
  </div>
</DocumentFragment>
`;
